panther_battery/src/roboteq_battery.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <panther_battery/roboteq_battery.hpp>
+
+#include <functional>
+#include <limits>
+#include <stdexcept>
+#include <vector>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <panther_utils/moving_average.hpp>
+
+namespace panther_battery
+{
+
+RoboteqBattery::RoboteqBattery(
+  const std::function<DriverStateMsg::SharedPtr()> & get_driver_state,
+  const RoboteqBatteryParams & params)
+: GetDriverState(get_driver_state), driver_state_timeout_(params.driver_state_timeout)
+{
+  voltage_ma_ = std::make_unique<panther_utils::MovingAverage<float>>(
+    params.voltage_window_len, std::numeric_limits<float>::quiet_NaN());
+  current_ma_ = std::make_unique<panther_utils::MovingAverage<float>>(
+    params.current_window_len, std::numeric_limits<float>::quiet_NaN());
+}
+
+bool RoboteqBattery::Present() { return true; }
+
+void RoboteqBattery::Update(const rclcpp::Time & header_stamp, const bool)
```

Comment:
```suggestion
void Update(const rclcpp::Time & header_stamp, const bool /* charger_connected */)
```
This shows what was the original use for the boolean variable and not cause unused error